File: session/02_double_dispatch/java?/Rock.java
public class Rock implements Beatable {
	@Override
	public String beats(Beatable other) {
		if (other.getClass() == Paper.class) {
			return "rock loses";
		}

		if (other.getClass() == Scissor.class) {
			return "rock wins";
		}
		return "draw";
	}
}

File: 02_double_dispatch/c#/01_before_pattern_matching/Rock.cs
public class Rock : Beatable {
	public string Beats(Beatable other) {
		if (other.GetType() == typeof(Paper)) {
			return "rock loses";
		}

		if (other.GetType() == typeof(Scissor)) {
			return "rock wins";
		}
		return "draw";
	}
}
